riemann_client/client.py

```diff
@@ -6,7 +6,11 @@
 
 from __future__ import absolute_import
 
+import logging
 import socket
+from threading import RLock
+from threading import Timer
```

Comment:
Not that I know of. You can check time elapsed every time a new record is passed to the client, but you might end up with messages sitting on the queue forever if no new records come around to trigger a new flush.